src/rpc/rawtransaction.cpp

```diff
@@ -1745,101 +1537,296 @@ UniValue signrawtransaction(const UniValue& params, bool fHelp)
             throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid sighash param");
     }
 
-    bool fHashSingle = ((nHashType & ~SIGHASH_ANYONECANPAY) == SIGHASH_SINGLE);
+    if ((txVersion == SC_CERT_VERSION) && (nHashType != SIGHASH_ALL)) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Unsupported sighash param for certificate");
+    }
 
-    // Script verification errors
-    UniValue vErrors(UniValue::VARR);
+#ifdef ENABLE_WALLET
+    else if ((txVersion != SC_CERT_VERSION) && pwalletMain)
```

Comment:
`(txVersion != SC_CERT_VERSION) &&` must be removed. Not it is fine for both certs and txs.